torchgeo/trainers/change.py

```diff
@@ -210,13 +266,35 @@ def _shared_step(self, batch: Any, batch_idx: int, stage: str) -> Tensor:
             x = rearrange(x, 'b t c h w -> b (t c) h w')
         y_hat = self(x)
 
-        loss: Tensor = self.criterion(y_hat, y.to(torch.float))
+        # For binary tasks, squeeze channel dimension to match expected shape
+        if self.hparams['task'] == 'binary':
+            y_hat = y_hat.squeeze(1)
+            y = y.squeeze(1)
+        elif self.hparams['task'] == 'multiclass':
+            if y.dim() == 4:  # If [batch, 1, H, W], squeeze to [batch, H, W]
```

Comment:
Will this if statement ever be false? I would like to ensure that the dimensions of all of our CD datasets are uniform.